profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/FirewallApiMockTest.java

```diff
@@ -0,0 +1,187 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jclouds.profitbricks.rest.features;
+
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import java.util.List;
+import org.apache.jclouds.profitbricks.rest.domain.FirewallRule;
+import org.apache.jclouds.profitbricks.rest.domain.options.DepthOptions;
+import org.apache.jclouds.profitbricks.rest.internal.BaseProfitBricksApiMockTest;
+import org.testng.annotations.Test;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+@Test(groups = "unit", testName = "FirewallApiMockTest", singleThreaded = true)
+public class FirewallApiMockTest extends BaseProfitBricksApiMockTest {
+   
+   @Test
+   public void testGeRuleList() throws InterruptedException {
+      server.enqueue(
+         new MockResponse().setBody(stringFromResource("/firewall/list.json"))
+      );
+      
+      List<FirewallRule> list = firewallApi().list("datacenter-id", "server-id", "nic-id");
+      
+      assertNotNull(list);
+      assertEquals(list.size(), 1);
+      assertEquals(list.get(0).properties().name(), "apache-firewall");
+      
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", "/datacenters/datacenter-id/servers/server-id/nics/nic-id/firewallrules");
+   }
+   
+   @Test
+   public void testGeRuleListWithDepth() throws InterruptedException {
+      server.enqueue(
+         new MockResponse().setBody(stringFromResource("/firewall/list.json"))
+      );
+      
+      List<FirewallRule> list = firewallApi().list("datacenter-id", "server-id", "nic-id", new DepthOptions().depth(3));
+      
+      assertNotNull(list);
+      assertEquals(list.size(), 1);
+      assertEquals(list.get(0).properties().name(), "apache-firewall");
+      
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", "/datacenters/datacenter-id/servers/server-id/nics/nic-id/firewallrules?depth=3");
+   }
+   
+   @Test
+   public void testGetRuleListWith404() throws InterruptedException {
+      server.enqueue(response404());
+      List<FirewallRule> list = firewallApi().list("datacenter-id", "server-id", "nic-id", new DepthOptions().depth(1));
+      assertTrue(list.isEmpty());
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", "/datacenters/datacenter-id/servers/server-id/nics/nic-id/firewallrules?depth=1");
+   }
```

Comment:
* Change the name to reflect that this tests the "withDepth" method.
* Still missing a test that verifies the list without depth and 404.